ledger/test-common/src/main/daml/benchtool/InterfaceSubscriptions.daml

```diff
@@ -0,0 +1,112 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module InterfaceSubscriptions where
```

Comment:
Following Simon’s suggestion, I’d change Foo2 to do 10 Foo2->FooData->Foo2 conversions and similarly 100 for Foo3. That gives us good idea of interpretation costs relative to data translation at the edges.

Happy to make a PR against your branch if you want to change that.

I think larger records are actually not all that interesting because then db deserialization also becomes more expensive so we expect the relative difference to stay more or less the same.

Reply:
sounds great! pls be my guest! 

my previous implementation scheme was heterogeneous FooX distribution. Do I get it correctly that after your change we need to have only one FooX template populated in the DB, and doing separate experiments with Foo2 and Foo3? (I think per create node CPU cost should stay constant to being able to explain the accumulated results)

Larger records: great point, let's make them minimal in the next experiment, so api and db serialization it is getting out of the picture! do you have an idea about the size?

Reply:
> Happy to make a PR against your branch if you want to change that.
you can commit directly to the branch as well

Reply:
> Do I get it correctly that after your change we need to have only one FooX template populated in the DB, and doing separate experiments with Foo2 and Foo3? (I think per create node CPU cost should stay constant to being able to explain the accumulated results)

I think what would work well is if you populate the db with equal numbers of Foo1, Foo2, Foo3 and then we compare subscriptions for Foo1, Foo2, Foo3. DB/value deserialization/serialization should be more or less identical between them (assuming you fill in the Text fields with equal size values) while interpretation overhead will be very different.